### DISCUSSION TITLE
Should've specify ActionScript version for different new feature?

### DISCUSSION THREAD

**Discussion by shaucky** [airsdk/Adobe-Runtime-Support#3136] [Ideas]
ActionScript 3.0 has a long term history. Developers know ActionScript **as** 3.0 all know it's born in 2006. But know we have some updates in language. Like new operators (`@""`, `?.`, `??`) and new basic data type (`float`).
 
Should we specify ActionScript 3.x for different feature? Like Python 3.x do. Now newest Python version is 3.13 (as i know), and Python 3.x use the same environment, it's just like what happens on ActionScript. Why don't we call ActionScript like ActionScript 3.1 and ActionScript 3.2? I think it's directly show people that ActionScript is still alive!
 
And the same time, the issue that Adobe Animate use a built-in compiler causes it can't compile new operators and literal (like `2.17f`). Through Adobe Animate calls the doc type ActionScript 3.0, i think it's also a way to show people diiference.

**Comment by ajwfrost** [comment]
Currently we're using the compiler version to determine what support is available. There are actually two things at play here:
- the compiler itself
- the runtime support

So for example, adding things like `@""` or `?.` are just changes within the compiler itself. They change how the compiler interprets tokens and turns those into code structures as it parses the AS3 file.. but they don't change the output of the ActionScript Byte Code. In contrast, the float support needed both a language update and a runtime update (hence the version number of the ActionScript Byte Code block will change if you use 'float', and older runtimes will not be able to load it).

If you run "mxmlc -version" you'll see the version of the compiler changing as we add those features; and we bumped up the major version when supporting the `float` type.

I guess we could start naming the ActionScript capabilities in a similar way with 3.0 being what we inherited from Adobe, and then incrementing a minor version when we added a new feature (I would still say that `float` is a minor version update; we can't really use "ActionScript 4" as a term, since there's already a specification about that one, and it would need a completely new virtual machine to be written...)

Agree re. Animate and the built-in compiler that it uses. Flex of course is similar and hasn't been updated for a while, but there is a Flex compiler within Apache Royale which is being updated, and we'd been looking to adapt that version as the one that is used in AIR too (but get a bit stuck internally when it comes to trying to update open source things..). 

thanks

**Reply by ajwfrost** [reply]
And, it looks like the documentation site could support this... so e.g. if we just took out the "3.0" in the title/path, and then started adjusting the version codes within the documentation source code, we could distinguish between things...
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6400804/d4bd256e-9f81-4fa2-b8e4-18ee9e19fd29)

**Reply by shaucky** [reply]
To be precise, my idea in this regard stems from **how to attract the interest of developers or beginners**. In fact, whether it is ActionScript 3.0 or 3.1 or likes, there is no clear correlation with the version number of ABC (46 to 3.0?). **ActionScript 3.0** is more like **a name** than something technical (relatively speaking).
 
As ordinary developers, we cannot simply refer to it (the language) as an unofficial name just because it has new syntax or basic data types when introducing it to others. This is definitely irresponsible. If there is a novel official statement, I think it is important for dissemination.

**Comment by 2jfw** [comment]
> I guess we could start naming the ActionScript capabilities in a similar way with 3.0 being what we inherited from Adobe, and then incrementing a minor version when we added a new feature

I like the idea and **ActionScript 3.1** (/ **3.x**) sounds good 👍
I never heard(/read) someone using the term **AS3.0** or **ActionScript 3.0**, it's always **AS3** or **ActionScript 3** so this would still fit and be valid when counting up the minor version

**Reply by Fancy2209** [reply]
AS2.0 and AS3.0 are the official names funnily enough

**Reply by mrfrasier** [reply]
*cries in ASNEXT*
